subsys/mpsl/fem/Kconfig

```diff
@@ -11,6 +11,11 @@ config MPSL_FEM_NRF21540_GPIO_SUPPORT
 	bool
 	default $(dt_nodelabel_has_compat,nrf_radio_fem,$(DT_COMPAT_NORDIC_NRF21540_GPIO))
 
+config MPSL_FEM_NRF21540_GPIO_SPI_SUPPORT
+	bool
+	depends on MPSL_FEM_NRF21540_GPIO_SUPPORT
+	default $(dt_nodelabel_has_prop,nrf_radio_fem,spi-if)
```

Comment:
Note: with this change in place, GPIO_SPI variant will become the default when used with devicetree overlays that contain full description of nRF21540, e.g. https://github.com/nrfconnect/sdk-nrf/blob/main/boards/shields/nrf21540_ek/nrf21540_ek.overlay

Reply:
Why?

This Kconfig option would default to y, but `CONFIG_MPSL_FEM_NRF21540_GPIO_SPI` still defaults to n - thus, it would not become the default implementation.

Reply:
The situation is slightly more complicated that either of us thought. When a devicetree overlay that contains full description of nRF21540 is applied, both:
`CONFIG_MPSL_FEM_NRF21540_GPIO_SUPPORT=y` and
`CONFIG_MPSL_FEM_NRF21540_GPIO_SPI_SUPPORT=y`.

In that case whether `CONFIG_MPSL_FEM_NRF21540_GPIO` or `CONFIG_MPSL_FEM_NRF21540_GPIO_SPI` is selected depends on the order they are put in `choice MPSL_FEM_CHOICE`. If you move `config MPSL_FEM_NRF21540_GPIO_SPI` to the top, it will become the default in such case. That's probably the desired behavior - otherwise manual setting of that choice is going to be necessary to enable GPIO_SPI variant, and so far we avoided that.

Reply:
Discussed F2F, decided to leave `CONFIG_MPSL_FEM_NRF21540_GPIO` as default even when SPI is defined in devicetree until `CONFIG_MPSL_FEM_NRF21540_GPIO_SPI` is no longer experimental.